### PR TITLE
Add option to change multiple ctrl values at once

### PR DESCRIPTION
**Pull request by leoratte**
Adds the option to change multiple control values at once with aqc_set_ctrl_vars. This new function accepts three arrays for val, offset and size and size of the arrays.

Use this approach instead of #43.

### CODE REVIEW

**Comment by aleksamagicka** [comment]
Thanks, will look into this ASAP.

File: aquacomputer_d5next.c [hunk 1]
```diff
@@ -556,6 +555,20 @@ static int aqc_get_ctrl_val(struct aqc_data *priv, int offset, long *val, size_t
 	return ret;
 }
 
+static int aqc_set_buffer_val(u8 buffer[], int offset, long val, size_t size)
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:558]: `u8 *buffer` to be consistent

File: aquacomputer_d5next.c [hunk 2]
```diff
@@ -1016,6 +1046,9 @@ static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 {
 	int ret, pwm_value, temp_sensor;
 	long ctrl_mode;
+	int offset[3];
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:1049]: Name these `ctrl_values_offsets`, `ctrl_values`, `ctrl_values_sizes` or similar, so it's clear what they are for. And add a comment above that they're for setting multiple values at once in the ctrl report.

File: aquacomputer_d5next.c [hunk 3]
```diff
@@ -567,16 +580,33 @@ static int aqc_set_ctrl_val(struct aqc_data *priv, int offset, long val, size_t
...
+
+	ret = aqc_send_ctrl_data(priv);
+
+unlock_and_return:
+	mutex_unlock(&priv->mutex);
+	return ret;
+}
+
+/* Refreshes the control buffer, updates values at offsets and writes buffer to device */
+static int aqc_set_ctrl_vals(struct aqc_data *priv, int offset[], long val[],
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:595]: As far as I can see `aqc_set_ctrl_val()` can be refactored to use this with size of 1? And I'd name the three array parameters in plural, they hold multiple values after all.

**Review by aleksamagicka — CHANGES_REQUESTED** [review]
Comments enclosed.

Side note: there's no need to force push every time. All commits will get squashed either way when I merge them and it's easier to look at incremental changes than re-read the whole thing each time and try to find the difference. :smiley:

**Comment by leoratte** [comment]
I've addressed all of your comments.

**Pull request merged**